pysc2/lib/features.py

```diff
@@ -516,6 +558,55 @@ def unit_vec(u):
         out["build_queue"] = np.stack(unit_vec(u)
                                       for u in ui.production.build_queue)
 
+    if self._feature_units:
+      def feature_unit_vec(u):
+        screen_pos = self._world_to_feature_screen_px.fwd_pt(point.Point.build(u.pos))
+        screen_radius = self._world_to_feature_screen_px.fwd_dist(u.radius)
+        return np.array((
+            # Match unit_vec order
+            u.unit_type,
+            u.alliance,  # Self = 1, Ally = 2, Neutral = 3, Enemy = 4
+            u.health,
+            u.shield,
+            u.energy,
+            u.cargo_space_taken, 
+            int(u.build_progress * 100),  # discretize
+
+            # Resume API order
+            int(u.health / u.health_max * 255) if u.health_max > 0 else 0,
+            int(u.shield / u.shield_max * 255) if u.shield_max > 0 else 0,
+            int(u.energy / u.energy_max * 255) if u.energy_max > 0 else 0,
+            u.display_type,  # Visible = 1, Snapshot = 2, Hidden = 3
+            u.owner,  # 1-15, 16 = neutral
+            screen_pos.x,
+            screen_pos.y,
+            u.facing,
+            screen_radius,
+            u.cloak,  # Cloaked = 1, CloakedDetected = 2, NotCloaked = 3
+            u.is_selected,
+            u.is_blip,
+            u.is_powered,
+            
+            # Not populated for enemies or neutral
+            u.cargo_space_max,
+            u.assigned_harvesters,
+            u.ideal_harvesters,
+            u.weapon_cooldown,
+        ), dtype=np.int32)
+    
+      raw = obs.raw_data
+    
+      if len(raw.units) > 0 and self._map_size is not None:
+        with sw("raw_data"):
+          # Update the camera location so we can calculate world to screen positions
+          self._update_camera(point.Point.build(raw.player.camera))
+          feature_units = []
+          for u in raw.units:
+            if u.is_on_screen and u.display_type != DisplayType.HIDDEN:
+              # If the unit is on screen and not hidden
```

Comment:
This comment is now just restating the actual condition so is redundant. It was useful when it was !=3, but not when it the right side has a semantic meaning.